pkg/abaputils/abaputils.go

```diff
@@ -142,10 +142,13 @@ func HandleHTTPError(resp *http.Response, err error, message string, connectionD
 		var abapErrorResponse AbapError
 		bodyText, readError := ioutil.ReadAll(resp.Body)
 		if readError != nil {
-			return readError
+			return err
 		}
 		var abapResp map[string]*json.RawMessage
-		json.Unmarshal(bodyText, &abapResp)
+		errUnmarshal := json.Unmarshal(bodyText, &abapResp)
+		if errUnmarshal != nil {
+			return err
```

Comment:
Again, why would `err` contain something meaningful here, if you have an error in `errUnmarshal`.
```suggestion
			return fmt.Errorf("failed to parse JSON response: %w", errUnmarshal)
```

Reply:
Same as above